src/Categories/Bicategory/Construction/LaxSlice.agda

```diff
@@ -0,0 +1,337 @@
+{-# OPTIONS --without-K --safe #-}
+
+-- Mentioned in passing https://ncatlab.org/nlab/show/slice+2-category
+
+open import Categories.Bicategory
+
+module Categories.Bicategory.Construction.LaxSlice
+       {o ℓ e t}
+       (𝒞 : Bicategory o ℓ e t)
+       where
+
+open import Categories.Enriched.Category
+open import Categories.Category renaming (Category to 1Category)
+open import Categories.Morphism.Reasoning
+
+open import Categories.Bicategory.Extras 𝒞
+
+open import Level
+
+record SliceObj (X : Obj) : Set (t ⊔ o) where
+  constructor sliceobj
+  field
+    {Y} : Obj
+    arr : Y ⇒₁ X
+
+module SliceHom (A : Obj) where
+
+  record Slice⇒₁ (X Y : SliceObj A) : Set (o ⊔ ℓ) where
+    constructor slicearr₁
+    private
+      module X = SliceObj X
+      module Y = SliceObj Y
+
+    field
+      {h} : X.Y ⇒₁ Y.Y
+      Δ   : X.arr ⇒₂ (Y.arr ∘ₕ h)
+
+  record Slice⇒₂ {X Y : SliceObj A} (H K : Slice⇒₁ X Y) : Set (ℓ ⊔ e) where
+    constructor slicearr₂
+    private
+      module Y = SliceObj Y
+      module H = Slice⇒₁ H
+      module K = Slice⇒₁ K
+
+    field
+      {ϕ} : H.h ⇒₂ K.h
+      E   : K.Δ ≈ (Y.arr ▷ ϕ ∘ᵥ H.Δ)
+
+  open hom.Equiv
+  open import Categories.Functor using (Functor)
+
+  _∘'_ : ∀ {X Y : SliceObj A}{H K L : Slice⇒₁ X Y} → Slice⇒₂ K L → Slice⇒₂ H K → Slice⇒₂ H L
+  _∘'_ {X}{Y}{H}{K}{L} (slicearr₂ {ϕ = ϕ} E) (slicearr₂ {ϕ = ψ} F) = slicearr₂ {ϕ = ϕ ∘ᵥ ψ} 
+     (begin
+      L.Δ ≈⟨ E ⟩
+      (Y.arr ▷ ϕ ∘ᵥ K.Δ) ≈⟨ refl⟩∘⟨ F ⟩
+      Y.arr ▷ ϕ ∘ᵥ (Y.arr ▷ ψ ∘ᵥ H.Δ) ≈˘⟨ hom.assoc ⟩
+      (Y.arr ▷ ϕ ∘ᵥ Y.arr ▷ ψ) ∘ᵥ H.Δ ≈⟨ (∘ᵥ-distr-▷ ⟩∘⟨refl) ⟩
+      Y.arr ▷ (ϕ ∘ᵥ ψ) ∘ᵥ H.Δ ∎
+        )
+    where module X = SliceObj X
+          module Y = SliceObj Y
+          module H = Slice⇒₁ H
+          module K = Slice⇒₁ K
+          module L = Slice⇒₁ L
+          open 1Category (hom X.Y A)
+          open HomReasoning
+          open Equiv
+          open import Categories.Morphism.Reasoning (hom X.Y A)
```

Comment:
All imports should go at the top of the file.

Often it's best to
```agda
import Categories.Morphism.Reasoning as MR
```
and then do `open MR (hom X.Y A)` at the point of use.